prov/efa/src/rxr/rxr_msg.c

```diff
@@ -234,7 +234,16 @@ ssize_t rxr_msg_post_rtm(struct rxr_ep *ep, struct rxr_op_entry *tx_entry, int u
 	assert(rtm_type >= RXR_REQ_PKT_BEGIN);
 
 	if (peer->is_local && ep->use_shm_for_tx) {
-		/* we know shm's capablity, so no need to check handshake */
+		/*
+		 * We know shm's capablity, so no need to check handshake.
+		 * AWS Neuron is currently not supported by the SHM provider.
+		 */
+
+		if (efa_mr_is_neuron(tx_entry->desc[0])) {
+			FI_WARN(&rxr_prov, FI_LOG_CQ,
+			"AWS Neuron is currently not supported by the SHM provider\n");
+			return -FI_EINVAL;
```

Comment:
This behavior seems different than before, why not go through the original code path (ie. outside this if block), if it's neuron.

Reply:
Not sure I understand the "original code path". So before this PR, we disable shm for all FI_HMEM in `rxr_ep_use_shm_for_tx`. But now we can only disable shm for certain FI_HMEM ifaces like neuron (and synapseai in the future). And it is expected that neuron applications or neuron-nccl will not use shm. So we should return an error in `rxr_msg_post_rtm` when posting rtm for shm. 

I don't know how we can move this code outside this if block?

Reply:
offline talked with @wckzhang. This patch changes the code's behavior. Before this PR, shm will be turned off silently if applications requests `FI_HMEM`. After this PR, shm will not be turned off silently, but when the application tries to send rtm with shm for neuron tx buffer, the code will error and fail.

This behavior change is by design after discussing with @wzamazon, because it is expected that neuron should not use libfabric for intra-node node communication.